### PR TITLE
User can finalize their order

### PR DESCRIPTION
**Pull request by lukeperrin10**
pt sotry: https://www.pivotaltracker.com/story/show/177903771
## User Story 
```
As a user
In order to inform the restaurant about my order
I would like to be able to finalize it
```
## Suggested changes
- adds function to do PUT request to /api/orders/:id with finalized tag in the body
- adds button to call this function
- adds order confirmation message with pickup time 30min after order was made
## Lessons Learned
- working with dates in JS is pain in the ass
## Video of test
https://recordit.co/ftomvWHDZC

### CODE REVIEW

File: cypress/fixtures/active_order.json [hunk 1]
```diff
@@ -1,48 +1,53 @@
 {
-  "message": "This item was added to your order",
+  "body": {"message": "This item was added to your order",
```
Comment (DavDan1) [inline cypress/fixtures/active_order.json:2]: Why did you add body?

File: src/components/ViewOrder.jsx [hunk 2]
```diff
@@ -47,6 +61,15 @@ class ViewOrder extends Component {
         {orderItems}
       </Container>
       <p data-cy='total-price'>Total price: {this.state.totalPrice}Kr</p>
+      <Button onClick={() => {this.finalizeOrder()}} data-cy="finalize-order-button">Confirm Order</Button>
```
Comment (DavDan1) [inline src/components/ViewOrder.jsx:64]: Looks good !

File: cypress/fixtures/active_order.json (hunk 1 above)
Comment (offcast-dev) [reply cypress/fixtures/active_order.json:2]: because we had to pass time stamps when order was updated and created and they have to be outside the block with message and items

File: src/modules/orderHelper.js [hunk 3]
```diff
@@ -1,7 +1,8 @@
 import axios from 'axios'
 
+const authHeaders = JSON.parse(localStorage.getItem('userData'))
```
Comment (DavDan1) [inline src/modules/orderHelper.js:3]: why did you change it to const?

File: src/components/ViewOrder.jsx [hunk 4]
```diff
@@ -1,12 +1,12 @@
 import React, { Component } from 'react'
-import {getOrder} from '../modules/orderData'
-import {Item, Container, Header} from 'semantic-ui-react'
+import {getOrder, closeOrder} from '../modules/orderHelper'
```
Comment (DavDan1) [inline src/components/ViewOrder.jsx:2]: why is it changed from /orderData to /orderHelper ?

File: src/modules/orderHelper.js (hunk 3 above)
Comment (offcast-dev) [reply src/modules/orderHelper.js:3]: Refactoring, this line was repeating 4 times in every function, so we made a separate const for it.

File: cypress/fixtures/active_order.json (hunk 1 above)
Comment (DavDan1) [reply cypress/fixtures/active_order.json:2]: thank you!

File: src/modules/orderHelper.js (hunk 3 above)
Comment (DavDan1) [reply src/modules/orderHelper.js:3]: Thanks, good job!

File: src/components/ViewOrder.jsx (hunk 4 above)
Comment (offcast-dev) [reply src/components/ViewOrder.jsx:2]: we decided to move all order-related functions to one single file

Comment (DavDan1) [reply src/components/ViewOrder.jsx:2]: Great idea!

File: src/components/ViewOrder.jsx [hunk 5]
```diff
@@ -20,14 +20,28 @@ class ViewOrder extends Component {
...
+    const d = new Date();
+    d.setHours(hh);
+    d.setMinutes(mm);
+    const pickUpTime = new Date(d.getTime() + 30 * 60 * 1000)    
+    this.setState({pickUpTime: `${pickUpTime.toString().slice(16, 21)}`})
+  }
+   
   
   render() {
     let orderItems = this.state.menuData.map((item, i) => {
```
Comment (arunbhalli) [inline src/components/ViewOrder.jsx:47]: can you explain this

File: src/components/ViewOrder.jsx [hunk 6]
```diff
@@ -20,14 +20,28 @@ class ViewOrder extends Component {
...
     })
   }
+
+  finalizeOrder = async () => {
+    let result = await closeOrder(this.props.orderId)
+    this.setPickUpTime(result.body.updated_at)
+  }
+
+  setPickUpTime = (timeOfOrder) => {
+    let timestamp =`${timeOfOrder.slice(11, 16)}`
```
Comment (arunbhalli) [inline src/components/ViewOrder.jsx:36]: can you explain this guys?

Comment (offcast-dev) [reply src/components/ViewOrder.jsx:36]: timeOfOrder is a string that contains full ISO date, yyyy-mm-dd ... etc. characters 11 to 16 have information on hours and minutes hh:mm, with slice method we read only this part of the string

Comment (arunbhalli) [reply src/components/ViewOrder.jsx:36]: thanks good explanation

**Review by arunbhalli — APPROVED** [review]
good work
